### ISSUE TITLE
Java scoring library is not loading model trained by sklearn

### ISSUE THREAD

**Issue by kfinley-tealium** [catboost/catboost#885]
Problem: Java scoring library is not loading model trained by sklearn
catboost version: 0.15.1, Java client version 0.2.5
Operating System: OSX 10.13.6
CPU: 2.2 GHz Intel Core i7

Training model code
```
import catboost as cb
import pandas as pd
x = pd.DataFrame(data={'a': [1, 3, 4, 99, 12],
                       'b': [0.5, 0, 1.3, 3, 44],
                       'c': [0.5, 0, 1.3, 0.91, 0],
                       'd': ['a', 'b', 'c', 'd', 'e']
                       })
y = pd.DataFrame(data={'y': [0, 1, 0, 1, 0]})
model = cb.CatBoostClassifier()
model.fit(x, y, cat_features=[3], verbose=False, plot=False)
proba = model.predict([1, 0.5, 0.5, 'c'], prediction_type='Probability')
print(proba)
[0.85163254 0.14836746]
model.save_model('./cb_model.bin')
```

Java code
```
CatBoostModel model = CatBoostModel.loadModel("./cb_model.bin");
CatBoostPredictions prediction = model.predict(new float[]{1f, 0.5f, 0.5f}, new String[]{"c"});
System.out.println(prediction.get(0, 0));
```
Fails on `loadModel` with error:
 ```
java.lang.ArrayIndexOutOfBoundsException: 3

	at ai.catboost.CatBoostJNIImpl.catBoostModelGetFeatureNames(Native Method)
	at ai.catboost.CatBoostJNI.catBoostModelGetFeatureNames(CatBoostJNI.java:62)
	at ai.catboost.CatBoostModel.loadModel(CatBoostModel.java:55)
```

**Comment by Allensmile** [comment]
I had the same problem, have you solved it? Thank you!

**Comment by kirkfin-tealium** [comment]
Unfortunately not. I'm assuming that the Java java prediction lib just isn't usable yet, as this seems like the most basic use case

**Comment by annaveronika** [comment]
We'll look into this, sorry for inconvenience!

**Comment by kfinley-tealium** [comment]
Thanks for looking into this issue. 
I was able to get past the model loading issue by rolling the Java library back to the `0.2.4` version. However, the prediction results from the Java api don't match the results from the python api.
Python predict (Probability) method returns `[0.85163254 0.14836746]`
Java predict method returns `-1.7474631374857454` 

Thanks again.

**Comment by kizill** [comment]
> Python predict (Probability) method returns `[0.85163254 0.14836746]`
> Java predict method returns `-1.7474631374857454`
Currently java predict raw formula value, without sigmoid (as `1 / (1 + exp(-(-1.7474631374857454))) == 0.1483674`, i'm currenlty adding method for probability postprocessing both for binclass and multiclass formulas in C API, so it'll also appear in JAVA wrapper.

**Comment by BraginIvan** [comment]
Found a problem.
It starts [here](https://github.com/catboost/catboost/blob/master/catboost/jvm-packages/catboost4j-prediction/src/main/java/ai/catboost/CatBoostModel.java#L55)
usedFeatureCount is lower than features number in the model.
In my case usedNumericFeatureCount was lower than real number of numeric features.
Moving deeper we come to [this code](https://github.com/catboost/catboost/blob/master/catboost/libs/model/model.cpp#L418)
so usedNumericFeatureCount ignores features "not UsedInModel" 
[UsedInModel is false when Borders is empty](https://github.com/catboost/catboost/blob/master/catboost/libs/model/features.h#L71)

Border is empty if feature can not be split.
There is a simple way to fix it for user, just remove all features with importance = 0  on training stage.

**Comment by cockroachzl** [comment]
> > Python predict (Probability) method returns `[0.85163254 0.14836746]`
> > Java predict method returns `-1.7474631374857454`
> 
> Currently java predict raw formula value, without sigmoid (as `1 / (1 + exp(-(-1.7474631374857454))) == 0.1483674`, i'm currenlty adding method for probability postprocessing both for binclass and multiclass formulas in C API, so it'll also appear in JAVA wrapper.

Could the above be put in the doc https://catboost.ai/docs/concepts/java-reference_catboostmodel__predict.html? I hit this issue but had no idea what went wrong until I saw the above.

**Comment by andrey-khropov** [comment]
Fixed by commit https://github.com/catboost/catboost/commit/8846e5c53a7db43d8068678527029429f52ab2c6, will be fixed in Maven in version 0.2.6.

Sorry for the delay.

**Comment by andrey-khropov** [comment]
Maven version 0.2.6 released today contains this fix.
